src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -430,32 +430,45 @@ class AssemblerConnectionManager(
 
         // prepare batch queries
         return connection.createStatement().use { stmt ->
-            authorizedPropertiesOfPrincipal.forEach { principal, propertyTypes ->
+            authorizedPropertiesOfPrincipal.forEach { (principal, propertyTypes) ->
                 val columns = (if (entitySet.isLinking) linkingEntityKeyIdColumnsList else entityKeyIdColumnsList) +
                         ResultSetAdapters.mapMetadataOptionToPostgresColumn(MetadataOption.ENTITY_KEY_IDS) +
                         propertyTypes.map { it.type.fullQualifiedNameAsString }
-                val grantSelectSql = grantSelectSql(tableName, principal, columns)
-                logger.info("AssemblerConnectionManager.grantSelectForEntitySet grant query: $grantSelectSql")
-                stmt.addBatch(grantSelectSql)
+                try {
+                    val grantSelectSql = grantSelectSql(tableName, principal, columns)
+                    logger.info("AssemblerConnectionManager.grantSelectForEntitySet grant query: $grantSelectSql")
+                    stmt.addBatch(grantSelectSql)
+                } catch (e: NoSuchElementException) {
+                    logger.error("Principal $principal does not exists but has $permissionsToCheck permission on entity set ${entitySet.id}")
+                }
             }
             stmt.executeBatch()
         }
     }
 
     fun grantSelectForEdges(stmt: Statement, tableName: String, entitySetIds: Set<UUID>): IntArray {
-        val permissions = EnumSet.of(Permission.READ)
+        val permissionsToCheck = EdmAuthorizationHelper.READ_PERMISSION
```

Comment:
Not necessary for this PR but I think this block(450-461) can be moved all the way up to `MaterializedEdgesProcessor` since its logic is independent of the work being done in this class to build and execute this SQL. That way we also don't have to wait for network calls + exceptions for each principal during the `.foreach` and can just pass `authorizedPrincipals` in from `MaterializedEdgesProcessor`

Reply:
-- though I don't remember the particular restrictions on what can/can't be done inside of `EntryProcessor`s

Reply:
Good idea! Will do!

Reply:
Should be fine as long as it is offloadable and read only iirc.